lib/FocusTrap.js

```diff
@@ -1,31 +1,91 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import ReactDOM from 'react-dom';
+import {addFocusListener, addBlurListener} from './focusListeners';
 
 class FocusTrap extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.focusListener;
+    this.blurListener;
+    this.willWrap = !this.props.noWrapper;
+    this.setupNativeListeners = this.setupNativeListeners.bind(this);
+    this.removeNativeListeners = this.removeNativeListeners.bind(this);
+  }
+
   static propTypes = {
     onFocus: PropTypes.func,
     onBlur: PropTypes.func,
     component: PropTypes.any,
+    noWrapper: PropTypes.bool,
     children: PropTypes.node
   };
 
   static defaultProps = {
     component: 'div'
   };
 
+  componentDidMount() {
+    if (!this.willWrap) {
+      this.setupNativeListeners();
+    }
+  }
+
+  componentWillUnmount() {
+    this.removeNativeListeners();
+  }
+
+  setupNativeListeners() {
+    if (!this.focusListener) {
+      const {
+        onFocus, 
+        onBlur, 
+      } = this.props;
+
+      const elem = ReactDOM.findDOMNode(this);
+
+      if (elem && !this.focusListener) {
+        if (!elem.getAttribute('tabindex')) {
+          elem.setAttribute('tabindex', '-1');
+        }
+        this.focusListener = addFocusListener(elem, onFocus);
+        this.blurListener = addBlurListener(elem, onBlur);
+      } else {
+        this.willWrap = true;
+      }
+    }
+  }
+
+  removeNativeListeners() {
+    if (this.focusListener) {
```

Comment:
It may make sense to record whether we add a new `tabindex` value on mount, so that we can remove it again on dismount to completely clear up after ourselves - unless this is always taken care of by the act of unmounting, anyway?

Reply:
I think that in order for a parent to unmount, the children would have to unmount as well, right? Is there a use case where <HotKeys> and <FocusTrap> would go away without doing an actual remount-from-scratch of the contained children? The code for this wouldn't be very complex, but I think we may need to wait for a specific need for this, unless you've got a specific one in mind.

Reply:
No, I think you are right that without exception, this would be taken care of anyway by all children being unmounted. I'm happy to leave this out for now.